### ISSUE TITLE
Field Export in Struct Output needs to be a pointer for omitempty to take effect

### ISSUE THREAD

**Issue by paulbes** [awslabs/goformation#294]
The PR https://github.com/awslabs/goformation/pull/291 added a struct for the `Output`, but the `Export` field is not a pointer, this means that we will get the serialised name of the struct:

```yaml
Outputs:
  PrivateSubnetIds:
    Export: {} <------------ As so
    Value:
      Value:
        Fn::Join:
        - ','
        - - Ref: PrivateSubnet00
          - Ref: PrivateSubnet01
          - Ref: PrivateSubnet02
```

When validating this template with cloud formation we end up getting the following error message:
```text
ValidationError: Template format error: Output PrivateSubnetIds is malformed. The Name field of every Export member is required.
```

To fix this I believe that `Export` must be a pointer, reproduction:

```go
package main

import (
	"fmt"
	"encoding/json"
)

type A struct {
	B  B `json:",omitempty"`
	C *C `json:",omitempty"`
} 

type B struct {}

type C struct {}

func main() {
	d, _ := json.Marshal(&A{})
	fmt.Println(string(d))
}
```

This gives the following output:

```text
{"B":{}}
```

Note that `C` is omitted from the output.

**Comment by xrn** [comment]
Hi @paulbes 

I am trying to understand the issue correctly but to be honest, is difficult for me. Could you help me reproduce the case? Because if this is indeed the issue and pointer is solution - we have to change in every place where json omitempty to pointer.

**Comment by PaulMaddox** [comment]
I do think moving this library to use pointers is the right direction (similar to how the AWS Go SDK uses `aws.String("string")`).

It will be a major breaking change, but so long as it is communicated properly I'm not against that. 

I don't have time to work on this at the moment though, but would happily accept a PR.

**Comment by paulbes** [comment]
This doesn't affect the basic types, only fields that refer to a struct, e.g.:

```go
package main

import (
	"encoding/json"
	"fmt"
)

type A struct {
	B       B      `json:",omitempty"`
	C       *C     `json:",omitempty"`
	String  string `json:",omitempty"`
	Integer int    `json:",omitempty"`
}

type B struct{}

type C struct{}

func main() {
	d, _ := json.Marshal(&A{})
	fmt.Println(string(d))
}
```

Will still return: `{"B":{}}`, as per the docs: https://golang.org/pkg/encoding/json/#Marshal

> The "omitempty" option specifies that the field should be omitted from the encoding if the field has an empty value, defined as false, 0, a nil pointer, a nil interface value, and any empty array, slice, map, or string.

Notice the point about nil pointers, so we only need to change the fields that are optional today, such as the `Export` field should be a pointer.

I can make a PR to fix the changes introduced in #291, but I probably don't have time for more than that.
